### PR TITLE
write failing test and fix truncate_scan_ranges fn

### PR DESCRIPTION
**Pull request by Oscar-Pepper**
Fixes: https://github.com/zingolabs/zingolib/issues/2244

fixes bug in scan range truncation where the predicate is reversed resulting in clearing the scan ranges instead of truncating to a given height

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -9473,7 +9473,7 @@ dependencies = [
 
 [[package]]
 name = "zingolib"
-version = "3.0.0-rc.1"
```
Comment (zancas) [inline Cargo.lock:9476]: Is this the correct version bump?  Should it not be to 3.0.1?

File: pepper-sync/src/sync/state.rs [hunk 2]
```diff
@@ -1067,3 +1067,30 @@ pub(super) fn update_found_note_shard_priority(
         );
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn truncate_scan_ranges() {
```
Comment (zancas) [inline pepper-sync/src/sync/state.rs:1076]: This seems like a good start, but I am wondering if this test can be expanded, or if we can have a suite of tests, where different boundary conditions are exercised for this fn?

For example, this test passes "250", why not add a suite of tests that hit a range of truncation points?   0,1,98,99,100,101

300 before truncation
300 after truncation

And then verify in separate assert statements with descriptive failure messages that the behavior is expected?

**Review by dorianvp — APPROVED** [review]

**Review by zancas — CHANGES_REQUESTED** [review]

File: Cargo.lock (hunk 1 above)
Comment (dorianvp) [reply Cargo.lock:9476]: That `*-rc.1` I think is invalid semver, and that's why cargo removed it automatically.

**Review by zancas — APPROVED** [review]
